content/en/docs/tasks/tools/install-kubectl.md

```diff
@@ -295,63 +295,111 @@ kubectl cluster-info dump
+    ~~~
+
+*Note that bash-completion sources all completion scripts in `/etc/bash_completion.d`.*
+
+Both approaches are equivalent. After restarting your shell, autocompletion should be working.
+
+#### Bash on macOS
+
+The default version of Bash on macOS is 3.2, and unfortunately, the kubectl completion script only works with Bash 4.1+. One way around this is to install a newer version of Bash on macOS (see instructions [here](https://itnext.io/upgrading-bash-on-macos-7138bd1066ba)).
+
+If you upgraded your version of Bash, you can install bash-completion with Homebrew:
+
+~~~bash
 brew install bash-completion@2
-```
+~~~
 
-Follow the "caveats" section of brew's output to add the appropriate bash completion path to your local .bashrc.
+*Note that the `@2` stands for bash-completion 2, which is required for the kubectl completion script. In turn, bash-completion 2 requires Bash 4.1+, that's why you needed to upgrade Bash.*
 
-If you installed kubectl using the [Homebrew instructions](#install-with-homebrew-on-macos) then kubectl completion should start working immediately.
+Follow the "Caveats" section of the `brew install` command. In particular add the following lines to your `~/.bashrc` or `~/.bash_profile` file:
 
-If you have installed kubectl manually, you need to add kubectl autocompletion to the bash-completion:
+~~~bash
+export BASH_COMPLETION_COMPAT_DIR=/usr/local/etc/bash_completion.d
+[[ -r /usr/local/etc/profile.d/bash_completion.sh ]] && . /usr/local/etc/profile.d/bash_completion.sh
+~~~
 
-```shell
-kubectl completion bash > $(brew --prefix)/etc/bash_completion.d/kubectl
-```
+Restart your shell and verify that bash-completion is correctly installed with `type _init_completion`.
 
-The Homebrew project is independent from Kubernetes, so the bash-completion packages are not guaranteed to work.
+You now need to take care of sourcing the kubectl completion script in your shell sessions. There are multiple ways in which you can do this:
 
-### Using Zsh
-If you are using zsh edit the ~/.zshrc file and add the following code to enable kubectl autocompletion:
+- Source the completion script in your `~/.bashrc` file:
 
-```shell
-if [ $commands[kubectl] ]; then
-  source <(kubectl completion zsh)
-fi
-```
+    ~~~bash
+    echo 'source <(kubectl completion bash)' >>~/.bashrc
+    ~~~
+- Add the completion script to `/usr/local/etc/bash_completion.d`:
 
-Or when using [Oh-My-Zsh](http://ohmyz.sh/), edit the ~/.zshrc file and update the `plugins=` line to include the kubectl plugin.
+    ~~~bash
+    kubectl completion bash >/usr/local/etc/bash_completion.d/kubectl
+    ~~~
+- If you installed kubectl with Homebrew, then the completion script should be already in `/usr/local/etc/bash_completion.d/kubectl`. In that case, you don't need to do anything.
+
+*Note that bash-completion (if installed with Homebrew) sources all the completion scripts in the directory that is set in the `BASH_COMPLETION_COMPAT_DIR` environment variable.*
```

Comment:
use the shortcode note as per the [style doc](https://kubernetes.io/docs/contribute/style/style-guide/#shortcodes)

Reply:
Done